### PR TITLE
Feat: Spinner

### PR DESCRIPTION
**Pull request by crishpeen**
The icon will be changed when we get it thru the Supernova Exporter

### CODE REVIEW

File: packages/web-twig/src/Resources/components/Spinner/Spinner.stories.twig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{% extends 'layout/default.html.twig' %}
+
+{% block content %}
+
+{# Alert Colors #}
+{% include '@components/Alert/stories/AlertColors.twig' %}
```
Comment (literat) [inline packages/web-twig/src/Resources/components/Spinner/Spinner.stories.twig:6]: You haven't run this code, have you? :D

File: packages/web-react/src/types/spinner.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { StyleProps, TextColorsDictionaryType, TransferProps } from './shared';
+
+export interface SpinnerProps extends StyleProps, TransferProps {}
+
+export interface SpiritSpinnerProps<T = void> extends SpinnerProps {
+  /** Color of the Spinner */
+  color?: TextColorsDictionaryType<T>;
```
Comment (literat) [inline packages/web-react/src/types/spinner.ts:7]: I will prefer to use `C` for Color.

```suggestion
  color?: TextColorsDictionaryType<C>;
```

**Review by literat — CHANGES_REQUESTED** [review]

File: packages/web-react/src/components/Spinner/__tests__/useSpinnerStyleProps.test.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+describe('useSpinnerStyleProps', () => {
+  it('should return defaults', () => {
+    const props = {};
+    const { result } = renderHook(() => useSpinnerStyleProps(props));
+
+    expect(result.current.classProps).toBe('animation-spin-clockwise');
+  });
+
+  it('should return primary color', () => {
```
Comment (tomassychra) [inline packages/web-react/src/components/Spinner/__tests__/useSpinnerStyleProps.test.ts:12]: How about making a loop on those colors?

Comment (adamkudrna) [reply packages/web-react/src/components/Spinner/__tests__/useSpinnerStyleProps.test.ts:12]: Or, the whole text color test could be abstracted and reused across all components using the text color dictionary. But that's surely for another issue 🙂.

File: packages/web-react/src/components/Spinner/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Spinner
+
+Spinner allows users to show loading state.
```
Comment (adamkudrna) [inline packages/web-react/src/components/Spinner/README.md:3]: Too much copy-paste… 🙈 

How about reusing (a simplified version of) what you have in `web`?

> The Spinner indicates something is loading.

File: packages/web-twig/src/Resources/components/Spinner/Spinner.stories.twig (hunk 1 above)
Comment (crishpeen) [reply packages/web-twig/src/Resources/components/Spinner/Spinner.stories.twig:6]: Damn :D

File: packages/web-react/src/types/shared/dictionaries.ts [hunk 5]
```diff
@@ -16,6 +17,11 @@ export type AlignmentXDictionaryType<T = undefined> = (typeof AlignmentX)[Alignm
 export type ActionColorsDictionaryKeys = keyof typeof ActionColors;
 export type ActionColorsDictionaryType<T = undefined> = (typeof ActionColors)[ActionColorsDictionaryKeys] | T;
 
+export type ActionLinkColorsDictionaryKeys = keyof typeof ActionLinkColors;
+export type ActionLinkColorsDictionaryType<T = undefined> =
+  | (typeof ActionLinkColors)[ActionLinkColorsDictionaryKeys]
+  | T;
```
Comment (literat) [inline packages/web-react/src/types/shared/dictionaries.ts:23]: I mean use the `C` in every type for Color.

**Review by literat — APPROVED** [review]
One little rename of types and you are good to go :-)

**Review by tomassychra — APPROVED** [review]

File: packages/web-react/src/types/shared/dictionaries.ts (hunk 5 above)
Comment (crishpeen) [reply packages/web-react/src/types/shared/dictionaries.ts:23]: Done, check it

**Pull request merged**